components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/listener/WorkflowAuditLogger.java

```diff
@@ -0,0 +1,159 @@
+import org.wso2.carbon.identity.workflow.mgt.exception.WorkflowException;
+
+import java.util.List;
+
+public class WorkflowAuditLogger extends AbstractWorkflowListener {
+
+    private static final Log AUDIT_LOG = CarbonConstants.AUDIT_LOG;
+    private static final String AUDIT_MESSAGE = "Initiator : %s | Action : %s | Data : { %s } | Result " +
+            ":  %s ";
+    private static final String AUDIT_SUCCESS = "Success";
+
+
+    /**
+     * Trigger after deleting the request
+     *
+     * @param workflowRequest
+     * @throws WorkflowException
+     */
+    @Override
+    public void doPostDeleteWorkflowRequest(WorkflowRequest workflowRequest) throws WorkflowException {
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String auditData = "\"" + "Request ID" + "\" : \"" + workflowRequest.getRequestId()
+                + "\"";
+        AUDIT_LOG.info(String.format(AUDIT_MESSAGE, loggedInUser, "Remove workflow request", auditData,
+                AUDIT_SUCCESS));
+    }
+
+    /**
+     * Trigger after delete the workflow
+     *
+     * @param workflow
+     * @throws WorkflowException
+     */
+    @Override
+    public void doPostDeleteWorkflow(Workflow workflow) throws WorkflowException {
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String auditData = "\"" + "Workflow ID" + "\" : \"" + workflow.getWorkflowId()
+                + "\"";
+        AUDIT_LOG.info(String.format(AUDIT_MESSAGE, loggedInUser, "Remove workflow", auditData, AUDIT_SUCCESS));
+    }
+
+    /**
+     * Trigger after adding a workflow
+     *
+     * @param workflowDTO
+     * @param parameterList
+     * @param tenantId
+     * @throws WorkflowException
+     */
+    @Override
+    public void doPostAddWorkflow(Workflow workflowDTO, List<Parameter> parameterList, int tenantId) throws
+            WorkflowException {
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String auditData = "\"" + "Workflow Name" + "\" : \"" + workflowDTO.getWorkflowName()
+                + "\",\"" + "Workflow Impl ID" + "\" : \"" + workflowDTO.getWorkflowImplId()
+                + "\",\"" + "Workerflow ID" + "\" : \"" + workflowDTO.getWorkflowId()
+                + "\",\"" + "Workflow Description" + "\" : \"" + workflowDTO.getWorkflowDescription()
+                + "\",\"" + "Template ID" + "\" : \"" + workflowDTO.getTemplateId()
+                + "\"";
```

Comment:
Thanks for pointing this, fixed with https://github.com/wso2/carbon-identity/pull/1675